manager/dispatcher/assignments.go

```diff
@@ -245,3 +260,24 @@ func (a *assignmentSet) message() api.AssignmentsMessage {
 
 	return message
 }
+
+// populateSecretFromPlugin populates the secret value for the given specification using the secret plugin subsystem.
+func (a *assignmentSet) populateSecretFromDriver(spec *api.SecretSpec, readTx store.ReadTx) error {
```

Comment:
Thanks @cpuguy83, the secret spec was populated by the calling method. I melded everything to a single function (`secret`, which fetch the value from raft store and populate the secret value if needed). To keep flow consistent, I've change the `Debug` message in case the secret is not found in raft-store to `Error`, hope this makes more sense now.